azurerm/internal/services/storage/tests/resource_arm_hpc_cache_test.go

```diff
@@ -0,0 +1,154 @@
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMHPCCacheDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMHPCCache_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHPCCacheExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMStreamAnalyticsJob_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_hpc_cache", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMHPCCacheDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMHPCCache_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHPCCacheExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMHPCCahce_requiresImport),
+		},
+	})
+}
+
+func testCheckAzureRMHPCCacheExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := acceptance.AzureProvider.Meta().(*clients.Client).Storage.CachesClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(ctx, resourceGroup, name)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on storageCacheCachesClient: %+v", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: HPC Cache %q (resource group: %q) does not exist", name, resourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMHPCCacheDestroy(s *terraform.State) error {
+	conn := acceptance.AzureProvider.Meta().(*clients.Client).Storage.CachesClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_hpc_cache" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		resp, err := conn.Get(ctx, resourceGroup, name)
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("HPC Cache still exists:\n%#v", resp)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMHPCCache_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctestvn-%d"
```

Comment:
could we use uppercase letters where possible
```suggestion
  name                = "acctest-VN-%d"
```